bluetooth: controller: Made BT_CTLR_RX_PRIO_STACK_SIZE non-hidden

mtpr-ot opened the pull request:
Added prompt to BT_CTLR_RX_PRIO_STACK_SIZE, allowing vendor specific
configuration of high priority Rx thread stack size.

Signed-off-by: Morten Priess <mtpr@oticon.com>

cvinayak commented:
@jhedberg need your comments here.

@mtpr-ot @ulfalizer can hidden options be overridden by SoC/Board defaults file?

cvinayak reviewed (changes requested):
As this stack size is purely subsystem defined, it should not be exposed to users. Can we check if there is other ways to use vendor specific values here without forking this file?

mtpr-ot commented:
> As this stack size is purely subsystem defined, it should not be exposed to users. Can we check if there is other ways to use vendor specific values here without forking this file?

Agree. Stack sizes are - as you know - highly architecture dependent, and should probably be defined elsewhere.

mtpr-ot commented:
> @jhedberg need your comments here.
> 
> @mtpr-ot @ulfalizer can hidden options be overridden by SoC/Board defaults file?

Unfortunately the board configuration cannot override a hidden option.

aescolar commented:
Adding @ulfalizer in case he would be able to provide some input

ulfalizer commented:
Don't know about the specifics here, but it's good to think of `prj.conf`, `defconfig`, etc., as being the same thing as going into the `menuconfig` and changing the value there. If you can't do it in `menuconfig`, you won't be able to do it from a configuration file either.

In the same vein, don't put a prompt on a symbol unless it would make sense for the user to go into the `menuconfig` and change the value there.

Might not be relevant here, but it's possible to make [prompt optional](https://docs.zephyrproject.org/latest/tools/kconfig/index.html#optional-prompts) by the way.

ulfalizer commented:
There's documentation on setting configuration values in [the board porting guide](https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#setting-configuration-values) as well by the way.

It should probably be moved somehwere else. Hard to find now. :/

aescolar commented:
@cvinayak @mtpr-ot : Would then an optional prompt be a good way forward?
Something like
```int "High priority Rx thread stack size"  if !SOC_COMPATIBLE_NRF```

cvinayak commented:
@aescolar @mtpr-ot I am ok with optional prompts.

mtpr-ot commented:
If everybody is OK with it, I'll make an optional prompt.

mtpr-ot committed:
bluetooth: controller: Made BT_CTLR_RX_PRIO_STACK_SIZE non-hidden

Added prompt to BT_CTLR_RX_PRIO_STACK_SIZE, allowing vendor specific
configuration of high priority Rx thread stack size for
!SOC_COMPATIBLE_NRF.

Signed-off-by: Morten Priess <mtpr@oticon.com>

cvinayak reviewed (approved):
LGTM

a1ien commented:
Why we guard by `if !SOC_COMPATIBLE_NRF`?
I want change BT_CTLR_RX_PRIO_STACK_SIZE in NRF 528400 board

mtpr-ot commented:
> Why we guard by `if !SOC_COMPATIBLE_NRF`?
> I want change BT_CTLR_RX_PRIO_STACK_SIZE in NRF 52840 board

Stack sizes are sub system dependent, "advanced parameters" trimmed for nRF boards, and hence not seen as something the user should need to deal with or change. If you really need this, please discuss with @cvinayak.
